### PR TITLE
Alpha submission

### PR DESCRIPTION
**Pull request by zakn-amd**
Adds:
- Infrastructure to support the alpha submission
  - GitHub Action run for Apptainer flow
- Documentation describing the alpha submission process

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,7 +131,7 @@ distclean: clean
 # Gradle is used to invoke the PartialRouterPhysNetlist class' main method with arguments
 # $< (%_unrouted.phys) and $@ (%_rwroute.phys), and display/redirect all output into %_rwroute.phys.log
 %_rwroute.phys: %_unrouted.phys | compile-java
-	(/usr/bin/time ./gradlew -DjvmArgs="$(JVM_HEAP)" -Dmain=com.xilinx.fpga24_routing_contest.PartialRouterPhysNetlist :run --args='$< $@') $(call log_and_or_display,$@.log)
+	(/usr/bin/time ./gradlew  -DjvmArgs="$(JVM_HEAP)" -Dmain=com.xilinx.fpga24_routing_contest.PartialRouterPhysNetlist :run --args='$< $@') $(call log_and_or_display,$@.log)
```
Comment (eddieh-xlnx) [inline Makefile:134]:
```suggestion
	(/usr/bin/time ./gradlew -DjvmArgs="$(JVM_HEAP)" -Dmain=com.xilinx.fpga24_routing_contest.PartialRouterPhysNetlist :run --args='$< $@') $(call log_and_or_display,$@.log)
```

File: alpha_submission/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Alpha Submission Containers
+
+This directory contains Example [Apptainer](https://apptainer.org/docs/user/latest/)
```
Comment (eddieh-xlnx) [inline alpha_submission/README.md:3]:
```suggestion
This directory contains example [Apptainer](https://apptainer.org/docs/user/latest/)
```

File: docs/alpha_submission.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+that specifies the base operating system image, and any further customisations
+(such as library installations) required to support an application. A
+definition file can then be compiled into an executable `*.sif` image which
+allows the application to be run in an isolated environment.
+
+The [contest repository](https://github.com/Xilinx/fpga24_routing_contest/)
+has been updated with example `*.def` files in the `alpha_submission` directory
+for both `rwroute` and `nxroute-poc`. To build and run the default container
+(which on a fresh clone would be `rwroute`):
+one would just run:
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:44]:
```suggestion
(which on a fresh clone would be `rwroute`) one would just run:
```

File: docs/alpha_submission.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+```
+make run-container
+```
+
+This is roughly equivalent to:
+```
+apptainer build rwroute_container.sif alpha_submission/rwroute_container.def
+apptainer run --pid --home `pwd` --rocm --bind /etc/OpenCL --mount src=/tools/,dst=/tools/,ro rwroute_container.sif make ROUTER="rwroute" BENCHMARKS=<list of all benchmarks>
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:53]:
```suggestion
apptainer run --pid --home `pwd` --rocm --bind /etc/OpenCL --mount src=/tools/,dst=/tools/,ro rwroute_container.sif make
```

File: docs/alpha_submission.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+* `--pid` runs the container in a new process ID namespace to ensure processes
+spawned by the container are not orphaned if the container is killed.
+* ``--home `pwd` `` sets the container home directory to be the current directory
+* `--rocm --bind /etc/OpenCL` configures [GPU Access](alpha_submission.md#gpu-access)
+* `--mount ...` creates a read-only mount of the host system's `/tools`
+directory to the container's `/tools` directory, which allows the container to
+access the host Vivado installation.
+
+The remainder of the Apptainer command line simply runs a make target in the
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:68]:
```suggestion
The remainder of the Apptainer command line simply runs the default make target from inside the
```

File: docs/alpha_submission.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+container.
+
+Finally, in order to aid in development the target
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:71]:
```suggestion
Finally, in order to aid in development the target:
```

File: docs/alpha_submission.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+```
+make test-container
+```
+
+has also been provided. This target is identical to the `run-container` target,
+except that results are only collected for the `boom_med_pb` benchmark, instead
+of collecting results for every benchmark. This allows contestants to quickly
+test their apptainer flow and avoid overloading shared resources should they
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:80]:
```suggestion
test their Apptainer flow and avoid overloading shared resources should they
```

**Review by clavin-xlnx — APPROVED** [review]

File: .github/workflows/test-container.yml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+          APPTAINER_VERSION: '1.2.5'
+        run: |
+          wget https://github.com/apptainer/apptainer/releases/download/v${APPTAINER_VERSION}/apptainer_${APPTAINER_VERSION}_amd64.deb
+          sudo dpkg -i apptainer_${APPTAINER_VERSION}_amd64.deb || true
+          sudo apt-get -f install
+      - run: |
+          make test-container
+      - uses: actions/upload-artifact@v3
+        with:
+          path: |
```
Comment (eddieh-xlnx) [inline .github/workflows/test-container.yml:30]:
```suggestion
          name: rwroute_container.sif
          path: |
```

File: docs/alpha_submission.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+* ``--home `pwd` `` sets the container home directory to be the current directory
+* `--rocm --bind /etc/OpenCL` configures [GPU Access](alpha_submission.md#gpu-access)
+* `--mount ...` creates a read-only mount of the host system's `/tools`
+directory to the container's `/tools` directory, which allows the container to
+access the host Vivado installation.
+
+The remainder of the Apptainer command line simply runs the default make target from inside the
+container.
+
+Finally, in order to aid in development the target:
```
[the remark this answers is not in the document]
Comment (zakn-amd) [reply docs/alpha_submission.md]: added clarification here: https://github.com/Xilinx/fpga24_routing_contest/pull/48/commits/3bb21788f36a16a6a80fc402fca0e5f280134300

File: docs/alpha_submission.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+* ``--home `pwd` `` sets the container home directory to be the current directory
+* `--rocm --bind /etc/OpenCL` configures [GPU Access](alpha_submission.md#gpu-access)
+* `--mount ...` creates a read-only mount of the host system's `/tools`
+directory to the container's `/tools` directory, which allows the container to
+access the host Vivado installation.
+
+The remainder of the Apptainer command line simply runs the default make target from inside the
+container.
+
+Finally, in order to aid in development the the Makefile target:
```
Comment (eddieh-xlnx) [inline docs/alpha_submission.md:70]:
```suggestion
Finally, in order to aid in development the Makefile target:
```

**Review by eddieh-xlnx — APPROVED** [review]

**Pull request merged**